TelemetryManager/Src/TMCircularBuffer.cpp

```diff
@@ -1,30 +1,56 @@
 
-#include "CircularBuffer.hpp"
+#include "TMCircularBuffer.hpp"
 
-TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(buf, size) {
+TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(*buf, size) {
     // Constructor
 }
 
-TMCircularBuffer::TMCircularBuffer(CircularBuffer buf): CircularBuffer(buf) {
+TMCircularBuffer::TMCircularBuffer(CircularBuffer* buf): CircularBuffer(*buf) {
     // Constructor
 }
 
 TMCircularBuffer::~TMCircularBuffer() {
     // Destructor
+   
+ }
+
+TMCircularBuffer::MAVLinkByte  TMCircularBuffer::dequeue() {
+    MAVLinkByte* res;
+    if(read(res,1)){
+        return *res;
+    }
+    return 0;
 }
 
-TMCircularBuffer::MAVLinkByte TMCircularBuffer::dequeue() { return 0; }
-
-void TMCircularBuffer::enqueue(MAVLinkByte byte) {
+bool TMCircularBuffer::enqueue(MAVLinkByte byte) {
     // Enqueue the byte
+    if(write(byte)){
+        index_ = (index_ + 1) % getSize();
+        return true;
+    }
+     return false;
 }
 
-int TMCircularBuffer::lastFullMessageEndIndex() {
+int TMCircularBuffer::bytesUntilLastMessageEnd(uint8_t* buf) {
```

Comment:
If you make buf in the common circular buffer protected, you should be able to access it in this method without requiring it as a parameter.

Reply:
You perhaps don't need `uint8_t* buf` parameter? Since you have access to the `buf_` now, you just read from buf_ in the function